public/app/features/plugins/built_in_plugins.ts

```diff
@@ -112,7 +113,7 @@ const builtInPlugins: any = {
   'app/plugins/panel/dashlist/module': dashListPanel,
   'app/plugins/panel/alertlist/module': alertListPanel,
   'app/plugins/panel/annolist/module': annoListPanel,
-  'app/plugins/panel/heatmap/module': heatmapPanel,
+  'app/plugins/panel/heatmap/module': config.featureToggles.useNewHeatmapPanel ? heatmapPanelNG : heatmapPanel,
```

Comment:
This is a bit dangerous, since the plugin will think of itself as `heatmap-new`... but seems to work 🤔

Reply:
This may be do to my lack of knowledge here, but could that end up applying old options to a new heatmap? Not sure what would happen in that case.

Reply:
yes, if we do not handle it properly :)

Each panel has a 'migration handler' that will get called, and we need to handle the migration for angular > react there.